### PR TITLE
Remove support for legacy RSpec matchers (pre 3)

### PR DESCRIPTION
**Pull request by pirj**
1. Remove support for legacy RSpec 1&2 matchers

As per http://rspec.info/blog/2013/07/the-plan-for-rspec-3/:

> compatibility layer so that existing matchers will continue to work, with the plan to remove that compatibility layer in RSpec 4

2. Remove deprecated `LegacyMacherAdapter` (with typo)

### CODE REVIEW

File: lib/rspec/expectations/configuration.rb [hunk 1]
```diff
@@ -86,10 +87,9 @@ def backtrace_formatter
       attr_writer :include_chain_clauses_in_custom_matcher_descriptions
 
       # Indicates whether or not custom matcher descriptions and failure messages
-      # should include clauses from methods defined using `chain`. It is
-      # false by default for backwards compatibility.
+      # should include clauses from methods defined using `chain`.
       def include_chain_clauses_in_custom_matcher_descriptions?
-        @include_chain_clauses_in_custom_matcher_descriptions ||= false
+        @include_chain_clauses_in_custom_matcher_descriptions
```
Comment (pirj) [inline lib/rspec/expectations/configuration.rb:92]: `||= true` obviously doesn't work when `false` is assigned explicitly.

**Review by JonRowe — APPROVED** [review]
👍 Minor change log tweak

File: Changelog.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ Breaking Changes:
 
 * Ruby < 2.3 is no longer supported. (Phil Pirozhkov, #1231)
 * Extract `should*` (both monkey-patching and non-monkey-patching). (Phil Pirozhkov, #1245)
+* Remove deprecated `LegacyMacherAdapter`. (Phil Pirozhkov, #????)
+* Remove support for legacy RSpec 1&2 matchers. (Phil Pirozhkov, #????)
```
[the remark this answers is not in the document]
Comment (pirj) [reply Changelog.md:9]: Oh, and I forgot the PR number.

**Comment by JonRowe** [comment]
Some how the last change has messed up the history to include all the other changes?

**Comment by pirj** [comment]
I've branched this PR off https://github.com/rspec/rspec-expectations/pull/1245, so this is as expected.

History needs some commit juggling anyway. I want to spread the Changelog changes over commits. Also, the base PR was fixed up, so a rebase on it will be needed.

Can't tell for sure, but I believe there might be some overlapping changes. Converted this to draft until #1245 is merged.

**Pull request merged**